snuba/query/extensions.py

```diff
@@ -25,15 +27,19 @@ class QueryExtension(ABC):
 
     def __init__(self,
         schema: Schema,
-        processor: ExtensionQueryProcessor,
+        processor: QueryProcessor[TExtensionPayload],
     ) -> None:
         self.__schema = schema
         self.__processor = processor
 
     def get_schema(self) -> Schema:
         return self.__schema
 
-    def get_processor(self) -> ExtensionQueryProcessor:
+    @abstractstaticmethod
+    def parse_payload(cls, payload: Mapping[str, Any]) -> TExtensionPayload:
+        raise NotImplementedError
```

Comment:
I think it would make sense to at some point couple this more with the schema (similar to how  `RequestSchema` works) rather than splitting that responsibility between the schema and this class.

Reply:
Yeah. I don't think we need to necessarily do it now, but I think coupling schema definition, parsing, and validation into one class would be a pattern generally worth following.